linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/FunctionEmitter.scala

```diff
@@ -2597,6 +2603,31 @@ private[emitter] class FunctionEmitter(sjsGen: SJSGen) {
         case GetClass(expr) =>
           genCallHelper("objectGetClass", transformExprNoChar(expr))
 
+        case Clone(expr) =>
+          val newExpr = transformExprNoChar(expr)
+          expr.tpe match {
+            /* If the argument is known to be an array, directly call its
+             * `clone__O` method.
+             */
```

Comment:
Maybe document that this is a theoretical case that only happens if `Clone` is used outside of `java.lang.Object` (which is allowed, it just tripped me off).

Reply:
It's not just theoretical. This happens every time we call `clone()` on an array, in fact. For example, in
```scala
val a: Array[Int] = ...
val b = a.clone()
```
the second statement as the following IR
```
val b: int[] = a.clone__O().asInstanceOf[int[]]
```
The optimizer sees `a.clone__O()` and, unlike previously where it was explicitly told *not* to inline it, it will now be able to inline it from `java.lang.Object.clone()`. Since it knows that the array type is a `Cloneable`, it simplifies it as
```scala
val b: int[] = <clone>(a).asInstanceOf[int[]]
```
and since the `<clone>` opcode is "polymorphic", in the sense that its result type is the type of its argument, the optimizer knows that the type of `<clone>(a)` is an `int[]` and can further simplify as
```scala
val b: int[] = <clone>(a)
```
This reaches the emitter, and falls into this case, which we eventually compile as
```javascript
var b = a.clone__O();
```

---

By comparison, before this commit, we prevented the inlining, and that meant that the `asInstanceOf` had to stay, resulting in the following JavaScript:
```javascript
var b = $asArrayOf_I(a.clone__O(), 1);
```